franken/ucontext/aarch64/swapcontext.S

```diff
@@ -1,35 +1,35 @@
 .global __franken_swapcontext
 .type __franken_swapcontext,#function
 __franken_swapcontext:
-	stp x19, x20, [x0,#0]
+	stp x19, x20, [x0,#0]   /* first buffer (prev) */
 	stp x21, x22, [x0,#16]
 	stp x23, x24, [x0,#32]
 	stp x25, x26, [x0,#48]
 	stp x27, x28, [x0,#64]
 	stp x29, x30, [x0,#80]
 	mov x2, sp
-	str x2, [x0,#104]
+	str x2, [x0,#104]       /* save return addr ptr for new sp */
 	stp  d8,  d9, [x0,#112]
 	stp d10, d11, [x0,#128]
 	stp d12, d13, [x0,#144]
 	stp d14, d15, [x0,#160]
+	str x30, [x0,#176]
```

Comment:
From [ARM Compiler armasm User Guide](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0801a/BABDEJIC.html)
> In AArch64 state, the Link Register (LR) stores the return address when a subroutine call is made. It can also be used as a general-purpose register if the return address is stored on the stack. The LR maps to register 30.

In here, I assume the host system ABI uses x30 as link register, so save the value to the context.